### PR TITLE
documentation for custom DataSource

### PR DESCRIPTION
**Pull request by virtualize**
Documentation on how to set the date time format for exported data.
As suggested here: https://github.com/sonata-project/SonataAdminBundle/issues/7105#issuecomment-824754776

### CODE REVIEW

File: docs/reference/data_source.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          $iterator->setDateTimeFormat('Y-m-d H:i:s');
+  
+          return $iterator;
+        }
+      }
+
+
+2. Add the your service to the services definition.::
+
+      # config/services.yaml
```
Comment (VincentLanglet) [inline docs/reference/data_source.rst:47]: A blank like seems needed after this line in order to please the linter

Comment (virtualize) [reply docs/reference/data_source.rst]: I've rephrased this.

**Comment by VincentLanglet** [comment]
cc @OskarStark Can you help about the linter errors ?

**Review by VincentLanglet — APPROVED** [review]

File: docs/reference/data_source.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  
+      use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+      use Sonata\AdminBundle\Exporter\DataSourceInterface;
+      use Sonata\DoctrineORMAdminBundle\Exporter\DataSource;
+      use Sonata\Exporter\Source\DoctrineORMQuerySourceIterator;
+      use Sonata\Exporter\Source\SourceIteratorInterface;
+  
+      class DecoratingDataSource implements DataSourceInterface
+      {
+        private DataSource $dataSource;
```
Comment (core23) [inline docs/reference/data_source.rst:26]:
```suggestion
          private DataSource $dataSource;
```

Please use 4 space indention for php example code

**Review by core23 — CHANGES_REQUESTED** [review]

File: docs/reference/data_source.rst (hunk 2 above)
Comment (virtualize) [reply docs/reference/data_source.rst]: Done

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks @virtualize

**Pull request merged**